### PR TITLE
Remove legacy network code

### PR DESCRIPTION
**Pull request by rphmeier**
This PR removes the legacy Polkadot-specific networking code, replacing it with the version introduced in #788 . This is the second-to-last step in removing `NetworkSpecialization` from the codebase.

The protocol handler was extended with certain additional capabilities to support collators.

### CODE REVIEW

**Comment by expenses** [comment]
Looks good as far as I can tell. tests need fixing.

File: availability-store/src/store.rs [hunk 1]
```diff
@@ -60,33 +60,31 @@ fn candidate_key(candidate_hash: &Hash) -> Vec<u8> {
...
-	*b"awaited_chunks"
-}
+const AWAITED_CHUNKS_KEY: [u8; 14] = *b"awaited_chunks";
 
 fn validator_index_and_n_validators_key(relay_parent: &Hash) -> Vec<u8> {
 	(relay_parent, 1i8).encode()
 }
 
+fn available_chunks_key(candidate_hash: &Hash) -> Vec<u8> {
+	(candidate_hash, 2i8).encode()
```
Comment (montekki) [inline availability-store/src/store.rs]: We could probably turn hese constants into an `enum` with explicit values? So we at least use named entities and may be sure that the values won't be used twice in two different places?

Comment (rphmeier) [reply availability-store/src/store.rs]: Yep, seems beyond the scope of the PR though. There are lots of other refactorings and weirdness in the availability store to deal with.

File: service/src/lib.rs [hunk 2]
```diff
@@ -572,6 +573,7 @@ where
 {
 	let inherent_data_providers = InherentDataProviders::new();
 
+	// TODO [now]: launch new network protocol.
```
Comment (montekki) [inline service/src/lib.rs]: The todo probably should be fixed before merging.

Comment (rphmeier) [inline service/src/lib.rs]:
```suggestion
```

Actually, it's unnecessary at the moment for a light client.

**Pull request merged**